packages/content-management/favorites/favorites_server/src/favorites_service.ts

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { SavedObject, SavedObjectsClientContract } from '@kbn/core-saved-objects-api-server';
+import { Logger, SavedObjectsErrorHelpers } from '@kbn/core/server';
+import { favoritesSavedObjectType, FavoritesSavedObjectAttributes } from './favorites_saved_object';
+
+export class FavoritesService {
```

Comment:
__question:__ do we have any strategy\plan to eventually clean up these `favorites` SO when the objects they references to are deleted?

Reply:
No plan. Here're the possible options I was thinking of https://github.com/elastic/kibana/pull/189285#issuecomment-2258553776, but they didn't seem better then leaving those stale ids in the objects. Curious, you have something in mind?

Reply:
I see, that makes sense to me, thanks! I don't have good ideas apart from a background cleanup job (e.g., we have one in Security for sessions that expired but were never explicitly invalidated, however, sessions are created much more frequently than user favorites, so the scale is slightly different) or a delete hook exposed by the Core for SO types that you could potentially subscribe to for cleaning up favorites for a specific SO type (doesn't exist yet AFAIK). 

I assume our telemetry will at least show if we have any favorites with hundreds of IDs, right? If we never reach, say, a threshold of 100 (not necessarily non-existent, but it would be helpful to understand the scale of a potential problem), then maybe it's not a big deal.

Reply:
> I assume our telemetry will at least show if we have any favorites with hundreds of IDs, right? If we never reach, say, a threshold of 100 (not necessarily non-existent, but it would be helpful to understand the scale of a potential problem), then maybe it's not a big deal.


yes, this is also what I am thinking 👍